db/util_testing.go

```diff
@@ -230,6 +230,11 @@ WHERE META(ks).xattrs._sync.sequence >= 0
 
 // ViewsAndGSIBucketReadier empties the bucket, initializes Views, and waits until GSI indexes are empty. It is run asynchronously as soon as a test is finished with a bucket.
 var ViewsAndGSIBucketReadier base.TBPBucketReadierFunc = func(ctx context.Context, b base.Bucket, tbp *base.TestBucketPool) error {
+	if c, ok := b.(*base.Collection); ok {
+		if err := c.DropAllScopesAndCollections(); err != nil && !errors.Is(err, base.ErrCollectionsUnsupported) {
```

Comment:
It seems that running in the GSI disabled case, `DropAllScopesAndCollections` would run 3 times: here, in `FlushBucketEmptierFunc` and in `collection.Flush` which could potentially cause test tear down to be slower. Shouldn't be a big issue because this teardown is done in the background, but could this cause some issues?

Unrelated to this change but for current behaviour, it seems every test tear down that uses the `FlushBucketEmptierFunc` runs `DropAllScopesAndCollections` twice: in `FlushBucketEmptierFunc` and then in `collection.Flush` (which `FlushBucketEmptierFunc` calls). Is this worth fixing in this PR too?

Reply:
Good catch - I reckon that moving this call to be after the `base.TestsDisableGSI()` check should get rid of one of those.

Reply:
Also removed the call in `Collection.Flush()` per Slack discussion.